### PR TITLE
Catch JSON parsing errors in json-to-messages

### LINKED ISSUE #992: json-to-messages: catch JSON parsing errors
## Summary

I'm trying to use `json-to-messages` as a library but I'm unable to trap JSON parsing errors, e.g.

```js
const readme = fs.createReadStream('README.md')
jsonToMessages(readme, process.stdout).
  catch(err => { console.error(`Trapped error: ${err}`) })
```

## Expected Behavior

I would expect the promise returned from `jsonToMessages` to be rejected with the error.

## Current Behavior

~~~
$ npm start

> node server.js

undefined:1
# Setting up your development environment
^

SyntaxError: Unexpected token # in JSON at position 0
    at JSON.parse (<anonymous>)
    at JSONTransformStream._flush (/Users/tooky/Code/json-to-messages-service/node_modules/@cucumber/json-to-messages/dist/src/stream/JSONTransformStream.js:30:24)
    at JSONTransformStream.prefinish (_stream_transform.js:140:10)
    at JSONTransformStream.emit (events.js:210:5)
    at prefinish (_stream_writable.js:645:14)
    at finishMaybe (_stream_writable.js:653:5)
    at endWritable (_stream_writable.js:673:3)
    at JSONTransformStream.Writable.end (_stream_writable.js:604:5)
    at IncomingMessage.onend (_stream_readable.js:691:10)
    at Object.onceWrapper (events.js:299:28)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! json-to-messages-service@0.0.1 start: `node server.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the json-to-messages-service@0.0.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tooky/.npm/_logs/2020-05-07T08_23_54_390Z-debug.log
 $ echo $?
1
~~~

An error is thrown internally but it exits the whole node process with error status 1.

### PR DESCRIPTION
**Pull request by tooky**
resolves #992

### CODE REVIEW

**Comment by tooky** [comment]
@vincent-psarga I'm not sure why the acceptance tests are failing? I think they may be failing on master.

**Comment by vincent-psarga** [comment]
Yes I still have issues getting the transformation to HTML or JSON to work on the CI. There are mainly meant for manual checking, that's why they don't make the CI fail.

File: json-to-messages/javascript/test/stream/JSONTransformStreamTest.ts [hunk 1]
```diff
@@ -19,4 +19,26 @@ describe('JSONReportStream', () => {
...
+    const data = '# this is not json'
+
+    const readableStream = new Stream.Readable({
+      read() {
+        this.push(data)
+        this.push(null)
+      },
+    })
+    const sink = new SingleObjectWritableStream<Array<any>>()
+    let f = () => {}
```
Comment (vincent-psarga) [inline json-to-messages/javascript/test/stream/JSONTransformStreamTest.ts:33]: `let f = () => null` mould fix the Linting problem iirc :)

**Comment by vincent-psarga** [comment]
Could you also update the [CHANGELOG](https://github.com/cucumber/cucumber/blob/master/json-to-messages/CHANGELOG.md) before we can merge that ? Thanks :)

**Comment by tooky** [comment]
@vincent-psarga hopefully that's good to go!

**Comment by tooky** [comment]
@vincent-psarga I missed that message somehow

CHANGELOG ✅

**Pull request merged**